src/features/island/buildings/components/DetailView.tsx

```diff
@@ -0,0 +1,152 @@
+import { OuterPanel } from "components/ui/Panel";
+import React from "react";
+
+import { ITEM_DETAILS } from "features/game/types/images";
+import Decimal from "decimal.js-light";
+import classNames from "classnames";
+import { secondsToString } from "lib/utils/time";
+import leftArrow from "assets/icons/arrow_left.png";
+
+import token from "assets/icons/token.gif";
+import timer from "assets/icons/timer.png";
+import lock from "assets/skills/lock.png";
+import { Button } from "components/ui/Button";
+import { BuildingName, BUILDINGS } from "features/game/types/buildings";
+import { GameState } from "features/game/types/game";
+
+interface Props {
+  state: GameState;
+  building: BuildingName;
+  onBack: () => void;
+  onBuild: (buildingName: BuildingName) => void;
+}
+
+export const DetailView: React.FC<Props> = ({
+  state,
+  building,
+  onBack,
+  onBuild,
+}) => {
+  const { bumpkin, inventory } = state;
+
+  const canBuild = (building: BuildingName) => {
```

Comment:
Should this function be called `cantBuild`

It is returning `return missingIngredients && missingBalance;` so assume the function returns true when you don't have the required ingredients